polybar-scripts/player-mpris-tail/player-mpris-tail.py

```diff
@@ -264,45 +264,34 @@ def _print(self, status):
-            else:
-                artists = _getProperty(self._metadata, 'xesam:artists', [''])
-                if artists != None and len(artists):
-                    # Note: This only grabs the first artist
-                    self.metadata['artist'] = re.sub(SAFE_TAG_REGEX, """\1\1""", artists[0])
-                else:
-                    self.metadata['artist'] = '';
-            self.metadata['album']  = re.sub(SAFE_TAG_REGEX, """\1\1""", _getProperty(self._metadata, 'xesam:album', ''))
-            self.metadata['title']  = re.sub(SAFE_TAG_REGEX, """\1\1""", _getProperty(self._metadata, 'xesam:title', ''))
-            self.metadata['track']  = _getProperty(self._metadata, 'xesam:trackNumber', '')
-            length = str(_getProperty(self._metadata, 'xesam:length', ''))
-            if not len(length):
-                length = str(_getProperty(self._metadata, 'mpris:length', ''))
-            if len(length):
-                # If we get a string formatted as a float, int(length) will ValueError, but int(float(length)) will not.
-                self.metadata['length'] = int(float(length))
-            else:
-                self.metadata['length'] = 0
-            self.metadata['genre']    = _getProperty(self._metadata, 'xesam:genre', '')
-            self.metadata['disc']     = _getProperty(self._metadata, 'xesam:discNumber', '')
-            self.metadata['date']     = re.sub(SAFE_TAG_REGEX, """\1\1""", _getProperty(self._metadata, 'xesam:contentCreated', ''))
-            self.metadata['year']     = re.sub(SAFE_TAG_REGEX, """\1\1""", self.metadata['date'][0:4])
-            self.metadata['url']      = _getProperty(self._metadata, 'xesam:url', '')
-            self.metadata['filename'] = os.path.basename(self.metadata['url'])
-            cover = _getProperty(self._metadata, 'xesam:artUrl', '')
-            if not len(cover):
-                cover = _getProperty(self._metadata, 'mpris:artUrl', '')
-            if len(cover):
-                self.metadata['cover'] = re.sub(SAFE_TAG_REGEX, """\1\1""", cover)
-            else:
-                self.metadata['cover'] = ''
-
-            self.metadata['duration'] = _getDuration(self.metadata['length'])
+            # Obtain properties from _metadata
+            _artist     = _getProperty(self._metadata, 'xesam:artist', [''])
+            _album      = _getProperty(self._metadata, 'xesam:album', '')
+            _title      = _getProperty(self._metadata, 'xesam:title', '')
+            _track      = _getProperty(self._metadata, 'xesam:trackNumber', '')
+            _genre      = _getProperty(self._metadata, 'xesam:genre', [''])
+            _disc       = _getProperty(self._metadata, 'xesam:discNumber', '')
+            _length     = _getProperty(self._metadata, 'xesam:length', 0) or _getProperty(self._metadata, 'mpris:length', 0)
+            _length_int = _length if type(_length) is int else int(_length)
```

Comment:
I think we still need `int(float(_length))` here instead of just `int(_length)`.

Reply:
@Cybolic, I've force-pushed the small change per your request.
 If I have to change something let me know.